### PR TITLE
chore_: extract message hash query for outgoing messages to go-waku

### PR DESCRIPTION
**Pull request by kaichaosun**
Extract message hash query for outgoing messages to go-waku

Important changes:
- [x] using channels from `MessageSentCheck` for sent and expired message events.

### CODE REVIEW

File: wakuv2/waku.go [hunk 1]
```diff
@@ -996,158 +985,17 @@ func (w *Waku) SkipPublishToTopic(value bool) {
...
-			w.logger.Debug("messages for next store hash query", zap.Any("messageIds", w.sendMsgIDs))
-			w.sendMsgIDsMu.Unlock()
-
-		}
-	}
-}
-
 func (w *Waku) ConfirmMessageDelivered(hashes []gethcommon.Hash) {
-	if !w.cfg.EnableStoreConfirmationForMessagesSent {
+	if !w.cfg.EnableStoreConfirmationForMessagesSent || w.messageSentCheck == nil {
```
Comment (chaitanyaprem) [inline wakuv2/waku.go]: Why do we need to check for both? 
If config is set, messageSentChecker should not be nil right.

File: wakuv2/waku.go [hunk 2]
```diff
@@ -1318,7 +1166,28 @@ func (w *Waku) Start() error {
 
 	go w.sendQueue.Start(w.ctx)
 
-	go w.checkIfMessagesStored()
+	if w.cfg.EnableStoreConfirmationForMessagesSent {
+		w.messageSentCheck = publish.NewMessageSentCheck(w.ctx, w.node.Store(), w.node.Timesource(), w.logger)
+		go w.messageSentCheck.CheckIfMessagesStored()
```
Comment (chaitanyaprem) [inline wakuv2/waku.go]: Maybe rename method to `Start()`

File: wakuv2/waku.go [hunk 3]
```diff
@@ -1318,7 +1166,28 @@ func (w *Waku) Start() error {
...
+		go func() {
+			for {
+				select {
+				case <-w.ctx.Done():
+					return
+				case hash := <-w.messageSentCheck.MessageStoredChan:
```
Comment (chaitanyaprem) [inline wakuv2/waku.go]: Why not return hash and state as a combo rather than having 2 channels?
Any specific reason

Comment (kaichaosun) [reply wakuv2/waku.go]: Need to create a new type to differentiate the EventEnvelopeSent/MessageStored and EventEnvelopeExpired/MessageExpired, and convert between types in status-go and go-waku, which seems not better than having two channels.

File: wakuv2/waku.go [hunk 4]
```diff
@@ -1340,7 +1189,29 @@ func (w *Waku) Start() error {
 
 	go w.sendQueue.Start(w.ctx)
 
-	go w.checkIfMessagesStored()
+	if w.cfg.EnableStoreConfirmationForMessagesSent {
```
Comment (chaitanyaprem) [inline wakuv2/waku.go:1192]: Maybe move everything under the if condition to a separate function? 
That way Start function would look leaner.

File: wakuv2/waku_test.go [hunk 5]
```diff
@@ -599,64 +597,6 @@ func waitForEnvelope(t *testing.T, contentTopic string, envCh chan common.Envelo
 	}
 }
 
-func TestConfirmMessageDelivered(t *testing.T) {
```
Comment (chaitanyaprem) [inline wakuv2/waku_test.go:602]: Hope we have an equivalent test in waku to validate the functionality.

**Review by chaitanyaprem — APPROVED** [review]
LGTM , few minor comments.

File: wakuv2/waku_test.go (hunk 5 above)
Comment (kaichaosun) [reply wakuv2/waku_test.go:602]: Equivalent test  is here: https://github.com/waku-org/go-waku/blob/master/waku/v2/api/publish/message_check_test.go

**Review by plopezlpz — APPROVED** [review]

**Pull request merged**